todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/addedittask/AddEditTaskFragment.java

```diff
@@ -78,30 +109,66 @@ public void onActivityCreated(Bundle savedInstanceState) {
 
         setTaskIdIfAny();
 
+        mViewModel = Injection.provideAddEditTaskViewModel(getContext(), mEditedTaskId);
+
         FloatingActionButton fab =
                 (FloatingActionButton) getActivity().findViewById(R.id.fab_edit_task_done);
         fab.setImageResource(R.drawable.ic_done);
         fab.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View v) {
                 if (isNewTask()) {
-                    mPresenter.createTask(
-                            mTitle.getText().toString(),
-                            mDescription.getText().toString());
+                    createTask();
```

Comment:
I would do it differently.
The better approach would be to have a method that returns Observable of the click events and use it in bind.